telluric/georaster.py

```diff
@@ -131,6 +209,7 @@ def _prepare_other_raster(one, other):
 
 
 def _merge(one, other, merge_strategy=MergeStrategy.UNION, requested_bands=None):
```

Comment:
this method looks to complex I think you need to split it to merge_intersection and merge_union, mainly because the interface is different,

If I understand correctly, in intersection you are expecting two single bands rasters, 

in union you expect rasters that share no band


and that is a inner method so we can do whatever we want

Reply:
I agree, let me split the method